flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -317,9 +318,13 @@ private RocksDBStateBackend(RocksDBStateBackend original, Configuration config)
 		// configure metric options
 		this.defaultMetricOptions = RocksDBNativeMetricOptions.fromConfig(config);
 
-		// copy remaining settings
-		this.predefinedOptions = original.predefinedOptions;
-		this.optionsFactory = original.optionsFactory;
+		// configure RocksDB predefined options
+		this.predefinedOptions = original.predefinedOptions == null ?
+			PredefinedOptions.valueOf(config.getString(RocksDBOptions.PREDEFINED_OPTIONS)) : original.predefinedOptions;
+
+		// configure RocksDB options factory
+		this.optionsFactory = original.optionsFactory == null ?
+			ConfigurableOptionsFactory.fromConfig(config) : original.optionsFactory;
```

Comment:
Here we could check if the user specified a different classname to choose a different `OptionsFactory` (other than `ConfigurableOptionsFactory`) through the config and not through code. Still, code should override config in this case.